mipengine/controller/api/error_handlers.py

```diff
@@ -32,7 +32,10 @@ def handle_bad_user_input(error: BadUserInput):
 
 @error_handlers.app_errorhandler(InsufficientDataError)
 def handle_privacy_error(error: InsufficientDataError):
-    ctrl_logger.getRequestLogger().info(f"Insufficient Data Error: \n " + error.message)
+    # TODO: Add proper context id. Related JIRA issue: https://team-1617704806227.atlassian.net/browse/MIP-486
+    ctrl_logger.get_request_logger("demoContextId123").info(
+        f"Insufficient Data Error: \n " + error.message
+    )
```

Comment:
https://team-1617704806227.atlassian.net/browse/MIP-486

Reply:
the only purpose of error_handlers.py module is (at least for now) to take care of what will be returned to the client calling the webapi. The contextId has no reason to exist in this module. ContextId is a property that should only exist in the Controller and AlgorithmExecution class, it should be irrelevant with the webapi whatsoever. The error_handlers module is part of the webapi component, not part of the Controller.The webapi is not part of what we call the controller layer, it is merely an API "in front" of the Controller.
The problem with the current implementation of the logger functionality originates from the assumption that the contextId property will be made available everywhere.

Reply:
unless off course we go forward with the abomination of adding a contextid everywhere..

Reply:
Understood. However, this is out of the scope of the tasks concerning this PR (please see:[https://team-1617704806227.atlassian.net/browse/MIP-311]). Based on the requirements, what should be done is to create a logger that can format logs to display the context id and also allow for the logger to display logs without it. The only reason you see this here, is for demonstrative purposes.

Since there are new tasks concerning the matter, I suggest we discuss this at the appropriate time, decide on the details then, and proceed with the logger changes that would then be final. If it is preferable, I could remove this altogether, even though I am not sure how this would resolve anything.

Reply:
I would suggest using better and more professional language. We shouldn't be offensive, as it does not add any value, just tension.

Reply:
It was not in my intentions to be offensive, the word "abomination" was humoristically oriented by my side and I m sorry if it sounded offensive.

Reply:
you can merge it, it is already approved